v1/partial/compressed.go

```diff
@@ -39,7 +40,58 @@ type CompressedImageCore interface {
 // Assert that Image is a superset of this partial interface.
 var _ CompressedImageCore = (v1.Image)(nil)
 
+// uncompressedImageExtender implements v1.Image by extending UncompressedImageCore with the
+// appropriate methods computed from the minimal core.
+type compressedImageExtender struct {
+	CompressedImageCore
+
+	lock     sync.Mutex
+	manifest *v1.Manifest
```

Comment:
These belong in the `compressImage` implementation in `v1/tarball`.  I suspect most other implementations don't need this.

Reply:
In fact, it turns out you have it :)